docs/contributing/testing.md

```diff
@@ -21,11 +21,18 @@ Fix any CSS or JavaScript style errors that the linting tests report.
 
 ## 2. Fix any invalid HTML
 
-To check whether your HTML markup is valid, [check your page with W3C's markup validator](https://validator.w3.org/).
+To make sure your HTML markup is valid, [check your page with W3C's markup validator](https://validator.w3.org/). The validator checks your document's source code. To see your document's source code, select [View Source](https://developer.mozilla.org/en-US/docs/Tools/View_source) in your browser.
 
-If you can, fix any errors or warnings the validator reports.
+You should also check your HTML markup is valid after JavaScript has modified it. To make sure, [check your page with W3C's 'Check serialized DOM of current page’ bookmarklet](https://validator.w3.org/nu/about.html#extras).
```

Comment:
Yeah, it's a standard thing for people who test HTML to know about.

Example of how JavaScript would modify a page:
Some components (for example, the accordion) require JS to work. When the page loads, if JS is enabled, then JS will change the page to add the component. Because JS is needed to show/hide content, loading without it can make hidden sections visible (among other ways it can mess with the intended display of a page).